modules/apps/commerce/commerce-channel-web/src/main/resources/META-INF/resources/js/paymentMethod_instalation/hooks/useProducts.ts

```diff
@@ -0,0 +1,247 @@
+					'filter',
+					`name eq 'Marketplace App Category'`
+				);
+
+				const response = await Liferay.Util.fetch(url.toString(), {
+					headers: {
+						...options?.headers,
+						'Content-Type': 'application/json',
+						'x-csrf-token': Liferay.authToken,
+					},
+				});
+
+				const data = await response.json();
+
+				if (data?.items?.length) {
+					return data;
+				}
+			}
+			catch (error) {
+				console.error('Failed to fetch channel ID:', error);
+			}
+			finally {
+				setLoading(false);
+			}
+		},
+		[]
+	);
+
+	const fetchChannelId = useCallback(
+		async (
+			options?: RequestInit
+		): Promise<APIResponse<Channel> | undefined | void> => {
+			setLoading(true);
+
+			try {
+				const url = new URL(
+					'https://marketplace-uat.liferay.com/o/headless-commerce-delivery-catalog/v1.0/channels'
+				);
+
+				url.searchParams.append('search', `name eq '${channelName}'`);
+
+				const response = await Liferay.Util.fetch(url.toString(), {
+					headers: {
+						...options?.headers,
+						'Content-Type': 'application/json',
+						'x-csrf-token': Liferay.authToken,
+					},
+				});
+
+				const data = await response.json();
+
+				if (data?.items?.length) {
+					return data;
+				}
+			}
+			catch (error) {
+				console.error('Failed to fetch channel ID:', error);
+			}
+			finally {
+				setLoading(false);
+			}
+		},
+		[]
+	);
+
+	const fetchProducts = useCallback(
+		async (
+			options?: RequestInit
+		): Promise<APIResponse<Product> | undefined | void> => {
+			if (!channelId) {
+				const getChannelResponse = await fetchChannelId();
+
+				setChannelId(getChannelResponse?.items[0]?.id);
+
+				return;
+			}
+
+			try {
+				const url = new URL(
+					`https://marketplace-uat.liferay.com/o/headless-commerce-delivery-catalog/v1.0/channels/${channelId}/products?accountId=-1&images.accountId=-1&nestedFields=productSpecifications,categories,images&filter=(categoryIds/any(x:(x+eq+'34083689')))`
+				);
+
+				url.searchParams.append('channelId', channelId.toString());
+				url.searchParams.append('pageSize', delta.toString());
+				url.searchParams.append('page', deltas.toString());
+				url.searchParams.append('sort', `name:${sort}`);
+
+				if (debouncedSearchQuery) {
+					url.searchParams.append('search', debouncedSearchQuery);
+				}
+
+				const response = await Liferay.Util.fetch(url.toString(), {
+					headers: {
+						...options?.headers,
+						'Content-Type': 'application/json',
+						'x-csrf-token': Liferay.authToken,
+					},
+				});
+
+				const data = await response.json();
+
+				setProductsResponse(data);
+			}
+			catch (error) {
+				console.error('Failed to fetch products:', error);
+			}
+			finally {
+				setLoading(false);
+			}
+		},
+		[channelId, debouncedSearchQuery, delta, deltas, fetchChannelId, sort]
+	);
+
+	useEffect(() => {
+		const handler = setTimeout(() => {
+			setDebouncedSearchQuery(searchQuery);
+		}, 1000);
+
+		return () => clearTimeout(handler);
+	}, [searchQuery]);
```

Comment:
Create or use a debounce hook, search for it in the portal or copy from Marketplace